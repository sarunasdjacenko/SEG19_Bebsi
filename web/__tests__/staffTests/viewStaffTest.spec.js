import { shallowMount } from "@vue/test-utils";
import Vue from "vue";
import VueRouter from "vue-router";
import db from "../../src/components/firebaseInit";
// @ts-ignore
import Component from "../../src/components/staff/ViewStaff.vue";
Vue.use(VueRouter);

describe("Component", () => {
  const wrapper = shallowMount(Component);
  test("is a Vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  }),
    //snapshot test
    test("renders correctly", () => {
      expect(wrapper.element).toMatchSnapshot();
    }),
    test("has the correct buttons rendered", () => {
      const addStaffBtn = wrapper.find("#addStaffBtn");
      const editBtn = wrapper.find("#editBtn");
      const deleteBtn = wrapper.find("#deleteBtn");
      expect(addStaffBtn).toBeDefined();
      expect(editBtn).toBeDefined();
      expect(deleteBtn).toBeDefined();
    }),
    test("test if data is set correctly", () => {
      const data = {
        email: "someEmail",
        name: "someName",
        dept: "someDep",
        role: "someRole"
      };
      // calls Vue.set recursively
      wrapper.setData({
        isAdmin: "Admin",
        currentEmail: "someEmail",
        users: [{ data }]
      });
      // Check if data() is set properly
      expect(wrapper.vm.isAdmin).toMatch("Admin");
      expect(wrapper.vm.currentEmail).toBeDefined();
      expect(wrapper.vm.users).toHaveLength(1);
    }),
    test("check if table is populated correctly", () => {
      // <td></td> exists
      expect(wrapper.find("td").exists()).toBe(true);
      // Check if all tds are rendered
      expect(wrapper.findAll("td")).toHaveLength(6);
      // Check if all ths are rendered
      expect(wrapper.findAll("th")).toHaveLength(5);
    }),
    test("check if delete method is triggered", () => {
      wrapper.setMethods({ deleteUser: jest.fn() });
      wrapper.find("#deleteBtn").trigger("click");
      expect(wrapper.vm.deleteUser).toHaveBeenCalled();
    });
  db.app.delete();
});
