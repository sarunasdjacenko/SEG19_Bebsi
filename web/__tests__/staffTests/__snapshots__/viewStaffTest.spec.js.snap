// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<div
  id="view-staff"
>
  <div
    class="container"
    style="width: 100%; height: 100%;"
  >
    <table
      class="collection with-header "
      style="background: white; width: 100%; height: auto;"
    >
      <thead
        class="collection-header"
      >
        <h4
          style="padding: 20px; font-size: 3em;"
        >
          <b>
            Staff
          </b>
        </h4>
         
        <tr
          style="font-size: 1.5em;"
        >
          <th
            style="padding: 20px;"
          >
            Role
          </th>
           
          <th>
            Email
          </th>
           
          <th>
            Name
          </th>
           
          <th>
            Department
          </th>
           
          <th />
        </tr>
      </thead>
        
      <router-link-stub
        class="btn green"
        event="click"
        id="addStaffBtn"
        style="margin: 20px;"
        tag="a"
        to="/register"
      >
        Add Staff
      </router-link-stub>
    </table>
  </div>
</div>
`;
